systemtests/src/test/java/io/enmasse/systemtest/common/upgrade/UpgradeTest.java

```diff
@@ -23,45 +33,172 @@
+
+        createUserCMD(kubernetes.getInfraNamespace(), "test-brokered", "test", "brokered", "v1alpha1");
+        createUserCMD(kubernetes.getInfraNamespace(), "test-standard", "test", "standard", "v1alpha1");
+        Thread.sleep(30_000);
+
+        createAddressCMD(kubernetes.getInfraNamespace(), "brokered-queue", "brokered-queue", "brokered", "queue", "brokered-queue", "v1alpha1");
+        createAddressCMD(kubernetes.getInfraNamespace(), "brokered-topic", "brokered-topic", "brokered", "topic", "brokered-topic", "v1alpha1");
+        createAddressCMD(kubernetes.getInfraNamespace(), "standard-queue", "standard-queue", "standard", "queue", "standard-large-queue", "v1alpha1");
+        createAddressCMD(kubernetes.getInfraNamespace(), "standard-topic", "standard-topic", "standard", "topic", "standard-small-topic", "v1alpha1");
+        createAddressCMD(kubernetes.getInfraNamespace(), "standard-anycast", "standard-anycast", "standard", "anycast", "standard-small-anycast", "v1alpha1");
+        createAddressCMD(kubernetes.getInfraNamespace(), "standard-multicast", "standard-multicast", "standard", "multicast", "standard-small-multicast", "v1alpha1");
+        Thread.sleep(30_000);
+
+        waitUntilDeployed(kubernetes.getInfraNamespace());
+        Thread.sleep(60_000);
+
+        assertTrue(sendMessage("brokered", new RheaClientSender(), new UserCredentials("test-brokered", "test"), "brokered-queue", "pepa", MESSAGE_COUNT, true));
+        assertTrue(sendMessage("standard", new RheaClientSender(), new UserCredentials("test-standard", "test"), "standard-queue", "pepa", MESSAGE_COUNT, true));
+
+        if (isAnsible) {
+            installEnmasseAnsible(Paths.get(Environment.getInstance().getUpgradeTemplates()), true);
+        } else {
+            upgradeEnmasseBundle(Paths.get(Environment.getInstance().getUpgradeTemplates()));
+        }
+        Thread.sleep(30_000);
+
+        if (!startVersion.equals("install")) {
+            // KW: I am seeing an issue in standard where after a Broker with queue depth comes up, sometimes messages
+            // don't flow down the link that the router has established. Sending a single message seems to unblock
+            // things.
+            assertTrue(sendMessage("standard", new RheaClientSender(), new UserCredentials("test-standard", "test"), "standard-queue", "pepa", 1, true));
+            assertTrue(receiveMessages("standard", new RheaClientReceiver(), new UserCredentials("test-standard", "test"), "standard-queue", 1, true));
+
+            assertTrue(receiveMessages("brokered", new RheaClientReceiver(), new UserCredentials("test-brokered", "test"), "brokered-queue", MESSAGE_COUNT, true));
+            assertTrue(receiveMessages("standard", new RheaClientReceiver(), new UserCredentials("test-standard", "test"), "standard-queue", MESSAGE_COUNT, true));
+        } else {
+            createUserCMD(kubernetes.getInfraNamespace(), "test-brokered", "test", "brokered", "v1alpha1");
```

Comment:
I don't really understand this branch.  I assume that it somehow deals with upgrade scenarios that lose messaging users.  Why do you only send messages in that case?  And `startVersion.equals("install")` seems a bit mystified.

Reply:
Because in downstream we lose users and messages after upgrade, I forgot to push my changes for this with fix for intall string which you mention later it is my bad